qdyn/output.f90

```diff
@@ -1195,6 +1283,148 @@ subroutine get_ivmax(pb)
 
 end subroutine get_ivmax
 
+!=====================================================================
+
+! CRP: Get the location of the maximum slip rate in each fault
+
+subroutine get_ivmax_fault(pb)
+
+  use problem_class
+  ! use my_mpi, only: is_MPI_parallel, sum_allreduce
+
+  type(problem_type), intent(inout) :: pb
+
+  integer, dimension(pb%nfault) :: ivmax_fault
+  double precision, dimension(pb%nfault) :: vmax_fault
+  ! double precision, dimension(1) :: buf
+  integer :: n, lbl, mesh_size, i_max_v
+  double precision :: max_v
+
+  ! Initialize ivmax_fault and vmax_fault to 0 for all faults
+  ivmax_fault = 0
+  vmax_fault = 0d0 
+  
+  ! mesh size
+  mesh_size=size(pb%v_glob)
+
+  do lbl =1, pb%nfault
+    max_v = 0d0
+    i_max_v = 0
+
+    do n=1, mesh_size
+      if (pb%mesh%fault_label_glob(n) == lbl) then
+        if (pb%v_glob(n) > max_v) then
+          max_v = pb%v_glob(n)
+          i_max_v = n 
+        end if
+      end if
+    end do
+
+    vmax_fault(lbl) = max_v
+    ivmax_fault(lbl) = i_max_v
+
+  end do
+
+
+  if (DEBUG) then
+    write(msg, *) "ivmax_fault=", ivmax_fault
+    call log_debug(msg, pb%it)
+    write(msg, *) "vmax_fault=", vmax_fault
+    call log_debug(msg, pb%it)
+    write(msg, *) "size pb%ivmax_fault=", size(pb%ivmax_fault)
+    call log_debug(msg, pb%it)
+    write(msg, *) "size pb%vmax_fault=", size(pb%vmax_fault)
+    call log_debug(msg, pb%it)
+  endif
+
+  ! Store the final vmax_fault array in pb%vmax_fault
+  pb%vmax_fault = vmax_fault
+
+  if (DEBUG) then
+    write(msg, *) " after storing pb%vmax_fault=", pb%vmax_fault
+    call log_debug(msg, pb%it)
+  endif
+
+
+    ! Store the final ivmax_fault array in pb%ivmax_fault
+  pb%ivmax_fault = ivmax_fault
+
+  if (DEBUG) then
+    write(msg, *) " after storing pb%ivmax_fault=", pb%ivmax_fault
+    call log_debug(msg, pb%it)
+  endif
+
+  ! ! Loop through all elements in the mesh per core
```

Comment:
Is the parallel implementation ready for production? Why is everything here commented out?

Reply:
I think it's because I'm retrieving the velocities and indices from the global variables instead, as in this part of the code:

```
do lbl =1, pb%nfault
    max_v = 0d0
    i_max_v = 0

    do n=1, mesh_size
      if (pb%mesh%fault_label_glob(n) == lbl) then
        if (pb%v_glob(n) > max_v) then
          max_v = pb%v_glob(n)
          i_max_v = n 
        end if
      end if
    end do

    vmax_fault(lbl) = max_v
    ivmax_fault(lbl) = i_max_v

  end do
```

Reply:
Ok, so all of the commented stuff below can be removed?

Reply:
yes, I removed it